src/main/java/util/Parser.java

```diff
@@ -0,0 +1,140 @@
+     * @throws DukeException
+     */
+    public static DukeCommand parse(String fullCommand) throws DukeException {
+        String commandParts[] = fullCommand.split(" ");
+        String command = commandParts[0];
+        try {
+            if (command.equals("list")) {
+                return new ListCommand("list");
+            } else if (command.equals("todo")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new ToDoCommand(fullCommand.substring(5));
+            } else if (command.equals("deadline")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new DeadlineCommand(fullCommand.substring(9));
+            } else if (command.equals("event")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new EventCommand(fullCommand.substring(6));
+            } else if (command.equals("mark")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new MarkCommand(fullCommand.substring(5));
+            } else if (command.equals("unmark")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new UnmarkCommand(fullCommand.substring(7));
+            } else if (command.equals("delete")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new DeleteCommand(fullCommand.substring(7));
+            } else if (command.equals("find")) {
+                Parser.exceptionCheck(commandParts.length, command);
+                return new FindCommand(fullCommand.substring(5));
+            } else if (command.equals("bye")) {
+                return new ByeCommand("bye");
+            } else {
+                throw new InvalidMessageException();
+            }
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    /**
+     * Checks if the user enters a command without the appropriate description
+     * @param size Length of the full user inputted command
+     * @param command User inputted String
+     * @throws DukeException
+     */
+    public static void exceptionCheck(int size, String command) throws DukeException {
+        if (size == 1) {
+            throw new EmptyDescriptionException(command);
+        }
+    }
+
+    public static String parseDescription(String description) {
+       return description.split("/")[0];
+    }
+
+    /**
+     * Generates and returns a LocalDateTime object from the user inputted description
+     * @param description The user inputted string description
+     * @param task The task by which a LocalDateTime should be generated for
+     * @return LocalDateTime object that
+     * @throws DukeException
+     */
+    public static LocalDateTime parseDateTime(String description, String task) throws DukeException {
+
+        String splitDesc[] = new String[0];
```

Comment:
I think a better naming convention should be `splitedDescription`, because now it sounds like a verb, which by convention is for method names.